kotlinpoet/src/commonMain/kotlin/com/squareup/kotlinpoet/jvm/alias/JvmTypeAliasKotlinPoetApi.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.kotlinpoet.jvm.alias
+
+/**
+ * An expected type for JVM to `typealias` only.
+ */
+@MustBeDocumented
+@Retention(AnnotationRetention.BINARY)
+@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY, AnnotationTarget.TYPEALIAS)
+@RequiresOptIn(
+  level = RequiresOptIn.Level.ERROR,
+  message = "An expected type for JVM to `typealias` only." +
+    " Don't implement or use it in non-JVM platforms.",
+)
+public annotation class JvmTypeAliasKotlinPoetApi
```

Comment:
Should we maybe call it `JvmOnlyKotlinPoetApi`? IMO this makes it a bit more clear that this type only exists in KotlinPoet for JVM.

Reply:
Sounds good!